packages/react-query/src/useBaseQuery.ts

```diff
@@ -29,29 +29,43 @@ export function useBaseQuery<
   >,
   Observer: typeof QueryObserver,
 ) {
+  // Duc - we need to create a queryClient and reuse it per query
+
+  // { context: options.context }
+  // Duc - why do we need context here?
+  // Duc - how to implement this behavior in angular
+  // Duc - FROM the document: Required, but only if no default query function has been defined See Default Query Function for more information.
+  // https://tanstack.com/query/v4/docs/reference/useQuery?from=reactQueryV3&original=https://react-query-v3.tanstack.com/reference/useQuery
+  // maybe we can skip this for angular e.g. the default query function is required
```

Comment:
context is needed to support multiple contexts :) see:

https://tanstack.com/query/v4/docs/guides/migrating-to-react-query-4#custom-contexts-for-multiple-providers

I don't know if angular has a similar concept. But it's not part of the query-core, so you can totally omit it.